modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -855,11 +855,15 @@ public String getName() {
         private int oldCount = 0;
 
         @Override protected void invalidated() {
+            int oldIndex = computeCurrentIndex();
```

Comment:
I wonder if we can cache the old index somewhere since the _compute..._ methods can take some time on big tables.

Reply:
I think an instance variable makes sense, as the current index is a property of a specific VirtualFlow instance.
That won't help with performance though, as we still need to compute the current index in a number of cases. Instead of calling the `computeCurrentIndex()` we would then just update the instance variable. 
To fix the issue, we need to reduce the cases where the current index is recomputed (or bring it to zero in case the current index can't theoretically change) , and this is what #1098 is about -- so your comment probably belongs to #1098?
I'll experiment with instance fields, and try to reduce the computations even more.

Reply:
Agreed. And good idea! I will also do a bunch of tests when I have time. :)